core/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -492,14 +492,9 @@ public static String termVectorOptionsToString(FieldType fieldType) {
         }
     }
 
-    protected static String indexTokenizeOptionToString(boolean indexed, boolean tokenized) {
-        if (!indexed) {
-            return "no";
-        } else if (tokenized) {
-            return "analyzed";
-        } else {
-            return "not_analyzed";
-        }
+    /* Only protected so that string can override it */
+    protected Object indexTokenizeOption(boolean indexed, boolean tokenized) {
+        return Boolean.valueOf(indexed);
```

Comment:
Boxing of primitive `boolean`s always leads to the same references as this is guaranteed by the JLS so the explicit call to `Boolean#valueOf` is not needed (however the OpenJDK implementation of the boxing conversion here is to just insert a call to `Boolean#valueOf` but that's just an implementation detail).

> If the value `p` being boxed is `true`, `false`, [...], then let `r1` and `r2` be the results of any two boxing conversions of `p`. It is always the case that `r1 == r2`.